Trailheads/src/classes/AnimalLocatorMock.cls

```diff
@@ -0,0 +1,15 @@
+@isTest
+global class AnimalLocatorMock implements HttpCalloutMock {
+    public static final String animalName = 'chicken';
+    global HttpResponse respond(HTTPRequest request) {
+        String animalId = request.getEndpoint().substringAfterLast('/');
+        AnimalLocator.Animal animal = new AnimalLocator.Animal(animalId, animalName, 'chicken food', 'cluck cluck');
+        System.debug('jkub // ' + request);
+        System.debug('jkub // ' + animalId);
+        HttpResponse response = new HttpResponse();
+        response.setHeader('Content-Type', 'application/json');
+        response.setBody('{"animal":' + JSON.serialize(animal) + '}');
```

Comment:
A tutaj wystarczyło zrobić mapę Map<String, Animal> i ją serializować, ale zatkało mi mózg wcześniej i na to nie wpadłem -.-'
Olej, chyba, że masz ochotę sprawdzić czy przejdzie serio